pykafka/test/kafka_instance.py

```diff
@@ -336,6 +378,133 @@ def produce_messages(self, topic_name, messages):
         return self.connection.produce_messages(topic_name, messages)
 
 
+class CertManager(object):
+    """Helper that generates key pairs/stores for brokers and clients
+
+    NB: intended only for test suite purposes.  For instance, currently the
+    same cert is shared between all brokers, and another between all clients,
+    even between different instances of CertManager.  And for another, this
+    class freely discloses passwords for its keys: they're purely here to
+    test if the configuration with passwords works
+    """
+
+    def __init__(self, root_dir):
+        # It would be more elegant to use a tempdir here, but that would make
+        # it difficult for KafkaConnection to find the CA cert for an already
+        # running cluster, and so we just use a fixed location:
+        self._dir = os.path.join(root_dir, "test-certs")
+
+        self.validity_days = '10000'
+
+        self.root_cert = os.path.join(self._dir, "root.cert")
+        self.root_key = os.path.join(self._dir, "root.key")
+        self.root_pass = "NfXI63pmNqbHF4tq"
+
+        self.truststore = os.path.join(self._dir, "broker.truststore.jks")
+        self.keystore = os.path.join(self._dir, "broker.keystore.jks")
+        self.broker_pass = "JJQqPLUV1ZIFD6ny"
+
+        self.client_cert = os.path.join(self._dir, "client.cert")
+        self.client_key = os.path.join(self._dir, "client.key")
+        self.client_pass = "dWJTxlilbJcH1Wgn"
+
+        if not os.path.exists(self.root_cert):
+            try:
+                os.makedirs(self._dir)
+            except OSError:
+                pass
+            self._gen_root_cert()
+            self._gen_broker_truststore()
+            self._gen_broker_keystore()
+            self._gen_client_cert()
+
+    def _gen_root_cert(self):
+        """Generate root certificate"""
+        cmd = ['openssl', 'req',
+               '-x509',
+               '-newkey', 'rsa:2048',
+               '-subj', '/CN=ca.example.com',
+               '-days', self.validity_days,
+               '-out', self.root_cert,
+               '-keyout', self.root_key,
+               '-passout', 'env:ROOT_PASS']
+        return subprocess.check_call(cmd, env=dict(ROOT_PASS=self.root_pass))
+
+    def _gen_broker_truststore(self):
+        """Generate truststore that trusts self.root_cert"""
+        cmd = ['keytool', '-importcert', '-alias', 'root',
```

Comment:
I'm not very well-versed in the details of SSL, so just making sure I understand here. These functions are necessary because the SSL dance requires the host machine to contain some specific files in its registry of trusted SSL entities?